### ISSUE TITLE
xCAT log into genesis shell after reboot

### ISSUE THREAD

**Issue by bhbai** [xcat2/xcat-core#1850]
Env: IBM Power Systems S822LC(8247-42L) with minimum firmware version OP810.10/OP8_v1.7_1.13

xCAT log into genesis shell after reboot
Expected behavior: Should not loginto genesis shell,the genesis should continue work to get Mac in CN.

=======

> Env
> lsxcatd -a
> Version 2.12 (git commit b33cb21, built Fri Jun 24 03:17:21 EDT 2016)
> This is a Management Node
> cfgloc=Pg:dbname=xcatdb;host=10.45.5.5|xcatadm
> dbengine=Pg
> dbname=xcatdb
> dbhost=10.45.5.5
> dbadmin=xcatadm
> [root@cit1055 ~]# lsb_release -a
> LSB Version: :core-4.1-noarch:core-4.1-ppc64le
> Distributor ID: RedHatEnterpriseServer
> Description: Red Hat Enterprise Linux Server release 7.2 (Maipo)
> Release: 7.2
> Codename: Maipo
> uname --a
> Linux 3.10.0-327.el7.ppc64le #1 SMP Thu Oct 29 17:31:13 EDT 2015 ppc64le ppc64le ppc64le GNU/Linux

```
[   89.594219] Freeing unused kernel memory: 576K (c000000001040000 - c0000000010d0000)
[   89.602632] systemd-udevd[5722]: starting version 208
Petitboot (dev.20151111)                                      8247-42L 06DF8E5
──────────────────────────────────────────────────────────────────────────────
  [Disk: sda2 / d2ad4e00-5d2b-4e43-9d34-35e5c8bc4f69]
    Red Hat Enterprise Linux Server (0-rescue-2f4a1512b6594baf8ad983cc56f5a934)
    Red Hat Enterprise Linux Server (3.10.0-327.el7.ppc64le) 7.2 (Maipo)
  [Network: eth0 / 00:90:fa:b9:ba:ae]
    xCAT
    xCAT

  System information
  System configuration
  Language
  Rescan devices
  Retrieve config from URL
*Exit to shell

──────────────────────────────────────────────────────────────────────────────
Enter=accept, e=edit, n=new, x=exit, l=language, h=help
The system is going down NOW!t
Sent SIGTERM to all processes
Sent SIGKILL to all processes
[   20.828863] be2net 0003:05:00.1: opcode 36-1 failed:status 68-0
[   26.425458] mlx4_en: eth7:   frag:0 - size:1518 prefix:0 align:0 stride:1536
[   26.638875] mlx4_en: eth7:   frag:0 - size:1518 prefix:0 align:0 stride:1536
[   26.657481] mlx4_en: eth9:   frag:0 - size:1518 prefix:0 align:0 stride:1536
[   26.669591] mlx4_en: eth10:   frag:0 - size:1518 prefix:0 align:0 stride:1536
[   26.679048] mlx4_en: eth11:   frag:0 - size:1518 prefix:0 align:0 stride:1536
[   26.689896] mlx4_en: eth12:   frag:0 - size:1518 prefix:0 align:0 stride:1536
[   26.698900] mlx4_en: eth13:   frag:0 - size:1518 prefix:0 align:0 stride:1536
[   26.709380] mlx4_en: eth14:   frag:0 - size:1518 prefix:0 align:0 stride:1536
Device "eth1" does not exist.
Cannot find device "eth1"
Device "eth2" does not exist.
Cannot find device "eth2"
Device "eth3" does not exist.
Cannot find device "eth3"
Device "eth4" does not exist.
Cannot find device "eth4"
Device "eth5" does not exist.
Cannot find device "eth5"
Device "eth6" does not exist.
Cannot find device "eth6"
Done
ERROR: BOOTIF missing, can't detect boot nic
ssh-keygen: generating new host keys: RSA1 RSA DSA ECDSA ED25519
Generating private key...Done
rsyslogd: error: option -c is no longer supported - ignored
Setting IP via DHCP...
Acquiring network addresses..[xCAT Genesis running on (none) /]# rsyslogd: error: option -c is no longer supported - ignored
```

**Comment by bhbai** [comment]
Note:
1. **Occasionally** will encounter this defect.
2. With the help of Raymond, we found the root cause, but hope you can fix it in 2.12.3 if possible.

> Raymond Zhao Sep 15, 2016 10:49 PM
> Found the root cause in the following code in **/opt/xcat/share/xcat/netboot/genesis/ppc64/fs/usr/bin/doxcat**:
> It will sleep 2 seconds if ip is not retrieved from server. The process will repeat 10 times.
> However, in our environment, genesis did not get dhcp ip in 2*10 seconds. then it fall into line 209: the shell mode.
> 
> The fix is to increase the sleep interval to bigger number like 10 seconds. and rebuid genesis.
> , please communicate with xCAT team to fix it 2.12.3


```
190             echo -n "Acquiring network addresses.."
191             tries=0
192             while [ -z "$bootnic" ]; do
193                 for tmp1 in $ALLUP_NICS; do
194                     if ip addr show dev $tmp1|grep -v 'scope link'|grep -v 'dynamic'|grep -v  inet6|grep inet > /dev/null; then
195                         result=`ping -c1 -I $tmp1 $XCATMASTER 2>&1`
196                         if [ $? -eq 0 ]; then
197                             bootnic=$tmp1
198                             break
199                         fi
200                 fi
201                 done
202                 sleep 2
203                 tries=$(($tries+1))
204                 if [ $tries -ge 10 ]; then
205                     break
206                 fi
207             done
208             if [ -z "$bootnic" ]; then
209                 /bin/bash
210             fi
```

**Comment by zet809** [comment]
Duplicated with #1638 , had been included in xcat 2.12.2 release.

**Comment by bhbai** [comment]
After replace with xcat 2.12.2 release(None MCP genesis: **CAT-genesis-base-ppc64-2.12-snap201607221057.noarch**) this issue still can be reproduced.

Expected behavior: Should not loginto genesis shell,the genesis should continue work.

 replace xCAT 2.12.2 rpm as following:

> 
> [root@cit1055 ~]# rpm -qa |grep -i xcat
> xCAT-genesis-base-x86_64-2.12-snap201607221057.noarch
> xCAT-buildkit-2.12.2-snap201609090313.noarch
> xCAT-server-2.12.2-snap201609090314.noarch
> xCAT-genesis-scripts-x86_64-2.12.2-snap201609090313.noarch
> conserver-xcat-8.1.16-10.ppc64le
> perl-xCAT-2.12.2-snap201609090312.noarch
> xCAT-vlan-2.12.2-snap201609090313.noarch
> xCAT-2.12.2-snap201609090314.ppc64le
> xCAT-client-2.12.2-snap201609090314.noarch
> xCAT-genesis-scripts-ppc64-2.12.2-snap201609090313.noarch
> ipmitool-xcat-1.8.15-3.ppc64le
> xCAT-genesis-base-ppc64-2.12-snap201607221057.noarch
